### ISSUE TITLE
[Bug?]: whileLoadingPage prop is not usable in Set/Private component

### ISSUE THREAD

**Issue by domvernon** [redwoodjs/redwood#6763]
### What's not working?

👋 I cannot seem to use the `whileLoadingPage` prop for a Set/Private component as indicated throughout the [Page loaders & PageLoadingContext](https://redwoodjs.com/docs/router#page-loaders--pageloadingcontext) section.

A workaround is to use the prop within each of the individual child Route components. Using `whileLoadingAuth` has no problems.

### How do we reproduce the bug?

```
<Private unauthenticated="signIn" roles="ADMIN" wrap={AppLayout} whileLoadingPage={SkeletonLoader}>
  <Route path="/profile" page={ProfilePage} name="profile" />
  <Route path="/dashboard" page={DashboardPage} name="dashboard" />
</Private>
```
Results in a type error within vscode, and when running it locally doesn't have the expected behaviour either.

### What's your environment? (If it applies)

```shell
System:
    OS: macOS 13.0
    Shell: 5.9 - /opt/homebrew/bin/zsh
  Binaries:
    Node: 16.15.0 - /private/var/folders/1m/mr7mmp951fg1qhpjf_1j_ddr0000gn/T/xfs-b0e6346e/node
    Yarn: 3.2.4 - /private/var/folders/1m/mr7mmp951fg1qhpjf_1j_ddr0000gn/T/xfs-b0e6346e/yarn
  Databases:
    SQLite: 3.39.4 - /usr/bin/sqlite3
  Browsers:
    Chrome: 106.0.5249.119
    Firefox: 106.0.1
    Safari: 16.1
  npmPackages:
    @redwoodjs/core: 3.1.2 => 3.1.2
```

### Are you interested in working on this?

- [ ] I'm interested in working on this

**Comment by jtoar** [comment]
Hey @domvernon, thanks for reporting this. I tried the example in the docs you linked to and even that's not working for me. Let me see if I can figure out what's going on

**Comment by cannikin** [comment]
Hey @Tobbe any quick idea what's going on here? Maybe `whilePageLoading` wasn't added to the `<Private>` component, but exists in `<Set>`? It definitely *should* work right?

/cc @thedavidprice Should we update the docs for now and put a notice that's it's not currently working for `<Private>` but that we're working on it?

**Comment by domvernon** [comment]
Hi @cannikin I don't actually think it is working for `<Set>` either given the two are related.

**Comment by Tobbe** [comment]
Looks like an oversight from my/our part. `whileLoadingPage` is not part of the code for Sets or Private at all right now. I wonder if we could just take `whileLoadingPage` and pass it along to all children...

I won't have time to look at this right now. But if anyone else is up for investigating if that's a viable route (har har) forward I'm happy to review/guide 🙂 

For now, a comment in the docs and possibly a link to this issue sounds good to me

**Comment by cannikin** [comment]
Instead of a bug maybe we should label this as a feature request!

**Comment by Tobbe** [comment]
Fixed by #6912
